src/server/generic_family.cc

```diff
@@ -851,7 +853,13 @@ void GenericFamily::Pexpire(CmdArgList args, ConnectionContext* cntx) {
   if (!absl::SimpleAtoi(msec, &int_arg)) {
     return cntx->SendError(kInvalidIntErr);
   }
-  int_arg = std::max<int64_t>(int_arg, 0L);
+  int_arg = std::max<int64_t>(int_arg, -1);
+
+  // to be more compatible with redis, we silently cap the expire time to kMaxExpireDeadlineSec
+  if (int_arg > kMaxExpireDeadlineSec * 1000) {
```

Comment:
Is there a reason to do this only in this fuctions? as we set expire in other commands such as SetEx